### PR TITLE
Replace the boundaries with last boundaries when importing cases in the network store

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

**Review by niconoir — APPROVED** [review]
There is a security hotspot to be reviewed (see SonarCloud results) which seems safe to me.

File: src/test/java/org/gridsuite/merge/orchestrator/server/CgmesCaseDataSourceClientTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    @Mock
+    private RestTemplate caseServerRest;
+
+    @Before
+    public void setUp() {
+    }
+
+    @Test
+    public void test() throws IOException, URISyntaxException {
+        UUID caseUuid = UUID.randomUUID();
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/merge/orchestrator/server/CgmesCaseDataSourceClientTest.java:51]: Rather use constant uuid ?

File: src/test/java/org/gridsuite/merge/orchestrator/server/CaseFetcherServiceTest.java [hunk 2]
```diff
@@ -113,4 +122,19 @@ public void test() {
...
         assertEquals("fileContent", new String(fileInfos.get(1).getData(), StandardCharsets.UTF_8));
     }
+
+    @Test
+    public void importCaseTest() {
+        List<BoundaryInfos> boundaries = new ArrayList<>();
+        boundaries.add(new BoundaryInfos("idBoundary1", "boundary1.xml", "fake content of boundary1"));
+        boundaries.add(new BoundaryInfos("idBoundary2", "boundary2.xml", "fake content of boundary2"));
+
+        UUID networkUuid = UUID.randomUUID();
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/merge/orchestrator/server/CaseFetcherServiceTest.java:132]: Rather use constant uuid ?

File: src/main/java/org/gridsuite/merge/orchestrator/server/CgmesBoundaryService.java [hunk 3]
```diff
@@ -46,32 +49,44 @@
...
             LOGGER.error("Error searching boundaries: {}", e.getMessage());
         }
-        return Collections.emptyList();
+        if (lastBoundaries.size() < 2) {
+            throw new PowsyblException("No last boundaries available !!!");
+        }
+        return lastBoundaries;
     }
 
+    public static List<FileInfos> getFileInfosFromLastBoundaries(List<BoundaryInfos> boundaryInfos) {
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/merge/orchestrator/server/CgmesBoundaryService.java:85]: getFileInfosFromBoundaries ?

File: src/test/java/org/gridsuite/merge/orchestrator/server/CgmesCaseDataSourceClientTest.java (hunk 1 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/merge/orchestrator/server/CgmesCaseDataSourceClientTest.java]: Done

File: src/test/java/org/gridsuite/merge/orchestrator/server/CaseFetcherServiceTest.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/merge/orchestrator/server/CaseFetcherServiceTest.java]: Done

File: src/main/java/org/gridsuite/merge/orchestrator/server/CgmesBoundaryService.java (hunk 3 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/merge/orchestrator/server/CgmesBoundaryService.java]: Done

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**